{
  ServletContext servletContext=getServletContext();
  servletContext.log("Exception while dispatching incoming RPC call",caught);
  try {
    response.setContentType("text/plain");
    response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
    response.getWriter().write(GENERIC_FAILURE_MSG);
  }
 catch (  IOException e) {
    servletContext.log("sendError() failed while sending the previous failure to the client",caught);
  }
}
